### PR TITLE
backport PR2264 into CHE-1818 branch

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?

Backport changes of PR 2264 into CHE-1818 workspace env refactoring
### What issues does this PR fix or reference?

Without this, change in master will be lost
### PR type
- [x] Minor change = no change to existing features or docs

### CODE REVIEW

**Comment by benoitf** [comment]
@garagatyi

**Comment by garagatyi** [comment]
I will merge CHE-1818 with squash. Are you ok if I'll squash theses changes too?

File: plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ComposeMachineProviderImpl.java [hunk 1]
```diff
@@ -162,6 +164,11 @@ public ComposeMachineProviderImpl(DockerConnector docker,
 
         allMachinesSystemVolumes = removeEmptyAndNullValues(allMachinesSystemVolumes);
         devMachineSystemVolumes = removeEmptyAndNullValues(devMachineSystemVolumes);
+
+        Set<String> volumes = new HashSet<>();
+        devMachineSystemVolumes.forEach((volume) -> Arrays.asList(volume.split(";")).stream().forEach((entry) -> volumes.add(entry)));
```
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ComposeMachineProviderImpl.java]: Consider usage of flatMap instead. I think that this operation can be clearer in that case.

Comment (skabashnyuk) [reply plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ComposeMachineProviderImpl.java]: what if?

```
        devMachineSystemVolumes = devMachineSystemVolumes.stream()
                                                         .map(line -> line.split(";"))
                                                         .flatMap(Arrays::stream)
                                                         .distinct()
                                                         .collect(Collectors.toSet());
```

**Comment by benoitf** [comment]
@garagatyi and @skabashnyuk I'm ok to squash everything or not use this PR at all as long as the change about handling multiple volumes with semi colon is handled.

@skabashnyuk as your solution is being more elegant, I've updated PR.

**Comment by garagatyi** [comment]
AFAIK now it is possible to change base branch of PR. So I think we can wait until CHE-1818 is merged and then change base branch of this PR to master and merge it to master. @benoitf WDYT?

**Comment by benoitf** [comment]
@garagatyi well current master is having the feature
so technically, merging it without merging first this into CHE-1818 will result in a regression
This is why I wanted to have it merged in what will be applied on master

**Comment by tolusha** [comment]
Why not to do the same for `allMachinesSystemVolumes` ?

**Comment by benoitf** [comment]
@tolusha I was not sure user was providing something for this variable

**Comment by tolusha** [comment]
It will be usefull for https://github.com/codenvy/codenvy/issues/499

**Comment by benoitf** [comment]
ok added

**Pull request merged**